src/mapbox/mapbox.js

```diff
@@ -236,13 +236,23 @@ export default class Mapbox {
       this._map._container = newContainer;
       Mapbox.savedMap = null;
 
-      // Update style
+      // Step3: update style and call onload again
+      const fireLoadEvent = () => props.onLoad({
+        type: 'load',
+        target: this._map
+      });
+
       if (props.mapStyle) {
-        this._map.setStyle(props.mapStyle);
+        this._map.setStyle(props.mapStyle, {
+          diff: true
+        });
+
+        // call onload event handler after style fully loaded when style needs update
+        (this._map.isStyleLoaded() ?
```

Comment:
This syntax is rather odd. Just use `if`?

Reply:
I got a warning of **Method '_create' has too many statements (26). Maximum allowed is 25.** if I use if statement, that’s why I choose this one syntax.

Reply:
You can add this line to the top of this method to disable the lint rule:

```js
/* eslint-disable max-statements */
```

I will refactor it after the patch.

Reply:
Yes, that's correct since it will overwrite the global settings. I can handle it by adding another commit.